core/src/main/java/io/questdb/cairo/sql/RecordCursorFactory.java

```diff
@@ -85,7 +93,11 @@ default boolean followedOrderByAdvice() {
      */
     RecordMetadata getMetadata();
 
-    default PageFrameCursor getPageFrameCursor(SqlExecutionContext executionContext) throws SqlException {
+    default PageFrameCursor getPageFrameCursor(SqlExecutionContext executionContext, int order) throws SqlException {
+        return null;
+    }
+
+    default PageFrameSequence<?> execute(SqlExecutionContext executionContext, Sequence collectSubSeq, int order) throws SqlException {
```

Comment:
ATM we don't use this public method anywhere, i.e. there is no code that calls `RecordCursorFactory#execute`. It's enough to keep it package protected in both `Async*Filtered*` factories. So, I'd better remove it from this interface until we have a good reason to introduce such public API. @bluestreak01 WDYT?

Reply:
@puzpuzpuz i think the plan was to hook up filter to concurrent aggregates pretty much right after. This is where this interface method comes in i think

Reply:
My point is that we don't need this API right now and it may confuse readers of this code. So, why don't we introduce it later, when we really need it? It's a matter of a few lines of code.

Reply:
i think so, this was to be a part of the API:

- record
- page frame
- async (this method)

Lets keep it please